src/services/getCosmosBalances.ts

```diff
@@ -0,0 +1,56 @@
+import { StargateClient } from "@cosmjs/stargate";
+import { ChainType, CosmosAddress, CosmosBalance, CosmosChain } from "../types";
+import { fromBech32, toBech32 } from "@cosmjs/encoding";
+
+const deriveCosmosAddress = (chainPrefix: string, address: string): string => {
+  return toBech32(chainPrefix, fromBech32(address).data);
+};
+
+export async function getCosmosBalances({
+  addresses,
+  cosmosChains
+}: {
+  addresses: CosmosAddress[];
+  cosmosChains: CosmosChain[];
+}): Promise<CosmosBalance[]> {
+  const cosmosBalances: CosmosBalance[] = [];
+
+  for (const chain of cosmosChains) {
+    if (chain.chainType !== ChainType.COSMOS) continue;
+
+    const addressData = addresses.find(
+      address => address.coinType === chain.coinType
+    );
+
+    if (!addressData) continue;
+
+    const cosmosAddress = deriveCosmosAddress(
+      chain.bech32Config.bech32PrefixAccAddr,
+      addressData.address
+    );
+
+    try {
+      const client = await StargateClient.connect(chain.rpc);
+      const balances = (await client.getAllBalances(cosmosAddress)) ?? [];
+
+      if (balances.length === 0) continue;
+
+      balances.forEach(balance => {
+        const { amount, denom } = balance;
+
+        cosmosBalances.push({
+          balance: amount,
+          denom,
+          chainId: String(chain.chainId),
+          decimals:
+            chain.currencies.find(currency => currency.coinDenom === denom)
+              ?.coinDecimals ?? 6
+        });
+      });
+    } catch (error) {
+      //
```

Comment:
left to return empty array

Reply:
After the try-catch block, it returns `cosmosBalances`, which is an empty array by default:

```js
const cosmosBalances: CosmosBalance[] = [];
```